### PR TITLE
Pratikmdhr/refactor frontend code

### PR DESCRIPTION
**Pull request by pratikmdhr**
## WHAT IT DOES:

_Description of the pull request, what changed_
Refactor front end code
- Moved api request code to top level component
- Added a separate slider component which can be used for three input types

## HOW TO TEST:

_Description of how to test if applicable_

### PREVIEW:

_Attach any relevant videos or images:_

---

### DESIGN:

_Attach link to figma file or design files here:_

---

### WHAT'S LEFT:

_Delete if not necessary:_

- [ ] - some task
- [ ] -

### CODE REVIEW

File: src/app/page.tsx [hunk 1]
```diff
@@ -4,16 +4,35 @@ import InputComponent from "~/components/Input/Input";
 import About from "~/components/About";
```
Comment (MaiaS) [inline src/app/page.tsx]: unused import?

File: src/app/page.tsx [hunk 2]
```diff
@@ -4,16 +4,35 @@ import InputComponent from "~/components/Input/Input";
...
   const [displayResult, setDisplayResult] = useState(false);
   const [currentResult, setCurrentResult] = useState<ResponseType | null>(null);
 
-  const handleSummarize = (newResult: ResponseType) => {
-    setLocalStorage(newResult);
-    setDisplayResult(true);
-    setCurrentResult(newResult);
+  const { isLoading, mutate, isError } = useMutation({
+    mutationFn: fetchSummaryData,
+    onSuccess: (res) => {
```
Comment (MaiaS) [inline src/app/page.tsx:20]: Nice refactor here!

File: src/components/Input/InputField/SummaryLengthSlider.tsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
+type SummaryLengthSliderPropType = {
+  summaryLength: string;
+  setSummaryLength: (length: string) => void;
+};
+
+const SummaryLengthSlider = ({ summaryLength, setSummaryLength }: SummaryLengthSliderPropType) => {
+  return (
```
Comment (MaiaS) [inline src/components/Input/InputField/SummaryLengthSlider.tsx:7]: Thank you for moving this to its own component!

**Review by MaiaS — APPROVED** [review]
Thank you for refactoring! Nice job cleaning up the code and separating out components!

File: src/app/page.tsx (hunk 1 above)
Comment (pratikmdhr) [reply src/app/page.tsx]: good catch! removed it and will add this later

**Pull request merged**